src/main/java/fi/tkgwf/ruuvi/config/Config.java

```diff
@@ -198,7 +201,15 @@ public static DBConnection getDBConnection() {
             case "dummy":
                 return new DummyDBConnection();
             default:
-                throw new IllegalArgumentException("Invalid storage method: " + storageMethod);
+                try {
+                    LOG.info("Trying to use custom DB connection class: " + storageMethod);
+                    if (customDBConnection == null) {
+                        customDBConnection = (DBConnection) Class.forName(storageMethod).newInstance();
```

Comment:
Should it really return a new instance for every call? Are parallel connections used somewhere in this application? Are you sure that using the same instance of `org.influx.impl.InfluxDBImpl` doesn't parallelize connections?

I actually thought of refactoring the existing database connections so that they would be consistent with this new way of creating only one instance, but didn't want to change that then -- at least not in the same commit where I added the possibility of a custom connection, and then I just forgot about it...

Reply:
`org.influx.impl.InfluxDBImpl` parallelizes TCP connections by default (which is what's used here), though while InfluxDB is not an issue, not all connection types necessarily parallelize the connections. The application doesn't yet make any use of parallel connections, but the major change I'm working on will most likely make use of that (multiple inputs (raw data sources) & multiple outputs (databases/storage))

Reply:
The configuration hell is exactly the reason why this change haven't materialized yet that much :) haven't figured out the best way to allow configuring such thing yet.

Though now that I think about it, the new "mimo" feature will likely need a separate class to manage all sorts of connections or something anyway, so maybe the getDBConnection could be refactored to return the same instance (with all connection types, for consistency), and then possibly deprecate that once the "mimo feature" is ready or something.

Reply:
Actually regarding this exact pull request we're discussing here, and the underlying issue #11, I still need to solve an issue not too dissimilar from yours. I'd like to be able to define the strategy per tag, so... I was thinking of adding new properties, either something like 

    limitingStrategy.default=AABBCCDDEE,1122334455
    limitingStrategy.onMovement=65432100000

or perhaps like this:

    tag.AABBCCDDEE.limitingStrategy=default
    tag.1122334455.limitingStrategy=default
    tag.65432100000.limitingStrategy=onMovement

Any thoughts? I think the latter format might be more intuitive and would support more per-tag properties more easily

Reply:
This is actually a good idea to implement, using the format of the latter example so that more per-tag properties could be easily implemented, for example different motion sensitivity thresholds for different tags, or even different update limits per-tag for the default strategy